sentence_transformers/sparse_encoder/trainer.py

```diff
@@ -0,0 +1,312 @@
+from __future__ import annotations
+
+import logging
+import os
+from typing import Any, Callable
+
+import torch
+from packaging.version import parse as parse_version
+from torch import nn
+from transformers import EvalPrediction, PreTrainedTokenizerBase, TrainerCallback
+from transformers import __version__ as transformers_version
+from transformers.integrations import WandbCallback
+
+from sentence_transformers.evaluation import SentenceEvaluator, SequentialEvaluator
+from sentence_transformers.sparse_encoder.data_collator import SparseEncoderDataCollator
+from sentence_transformers.sparse_encoder.losses import (
+    SparseMultipleNegativesRankingLoss,
+)
+from sentence_transformers.sparse_encoder.model_card import SparseEncoderModelCardCallback
+from sentence_transformers.sparse_encoder.SparseEncoder import SparseEncoder
+from sentence_transformers.sparse_encoder.training_args import (
+    SparseEncoderTrainingArguments,
+)
+from sentence_transformers.trainer import SentenceTransformerTrainer
+from sentence_transformers.util import is_datasets_available, is_training_available
+
+if is_datasets_available():
+    from datasets import Dataset, DatasetDict, IterableDataset, Value
+
+logger = logging.getLogger(__name__)
+
+
+class SparseEncoderTrainer(SentenceTransformerTrainer):
+    # TODO: Check if there is no other things we need to overwrite
```

Comment:
You'll definitely need to override the `_resume_from_checkpoint`. Other than that, it's not a lot